frontend_app_html-document/src/container/HtmlDocument.jsx

```diff
@@ -846,7 +849,18 @@ export class HtmlDocument extends React.Component {
           i18n={i18n}
         >
           <div> {/* this div in display flex, justify-content space-between */}
-            <div className='d-flex'>
+            <ToolBar>
+              <FavoriteButton
+                favoriteState={props.isContentInFavoriteList(state.content, state)
+                  ? FAVORITE_STATE.FAVORITE
+                  : FAVORITE_STATE.NOT_FAVORITE}
+                onClickAddToFavoriteList={() => props.addContentToFavoriteList(
+                  state.content, state.loggedUser, this.setState.bind(this)
+                )}
+                onClickRemoveFromFavoriteList={() => props.removeContentFromFavoriteList(
+                  state.content, state.loggedUser, this.setState.bind(this)
+                )}
+              />
```

Comment:
Maybe make this a component so the code can be shared across the apps? it's okay if you don't, the whole <Toolbar> will probably need to be factored later anyway

Reply:
This would be feasible, I'm just unsure of the real gain since `this.props`,`this.state` and `this.setState` are all used here.

Reply:
I let you decide, do what seems best.

Reply:
I left it as it is, I'm not inspired…